Bottom: 7292817ed24068611ad6cac825c4f26c8476920e
Top:    43f639f6fd5107bf4b50f4200b055136fa2925b5
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-05-21 16:39:46 -0700

rstream: Check for connection error on async connect

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/examples/rstream.c b/examples/rstream.c
index 104b318..d2f4a9b 100644
--- a/examples/rstream.c
+++ b/examples/rstream.c
@@ -448,7 +448,8 @@ static int client_connect(void)
 {
 	struct addrinfo *res;
 	struct pollfd fds;
-	int ret, rs;
+	int ret, rs, err;
+	socklen_t len;
 
  	ret = getaddrinfo(dst_addr, port, NULL, &res);
 	if (ret) {
@@ -477,6 +478,17 @@ static int client_connect(void)
 		ret = do_poll(&fds);
 		if (ret)
 			goto err;
+
+		len = sizeof err;
+		ret = getsockopt(rs, SOL_SOCKET, SO_ERROR, &err, &len);
+		if (ret)
+			goto err;
+		if (err) {
+			ret = -1;
+			errno = err;
+			perror("async rconnect")
+			goto err;
+		}
 	}
 
 free:
